### PR TITLE
feat: oauth providers

### PR DESCRIPTION
**Pull request by iskhakov**
## Summary

This PR implements OAuth authentication support for MCP (Model Context Protocol) servers, enabling secure third-party integrations with Google and Slack services. The implementation uses the OAuth 2.0 PKCE (Proof Key for Code Exchange) flow for enhanced security.

### CODE REVIEW

File: desktop_app/openapi/archestra/catalog/openapi-ts.config.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ const __filename = fileURLToPath(import.meta.url);
 const __dirname = path.dirname(__filename);
 
 export default defineConfig({
-  input: 'https://www.archestra.ai/mcp-catalog/api/docs',
+  input: 'http://localhost:3000/mcp-catalog/api/docs',
```
Comment (iskhakov) [inline desktop_app/openapi/archestra/catalog/openapi-ts.config.ts]: TODO: make this configurable

**Comment by iskhakov** [comment]
missing tests: Missing Test Coverage for OAuth Implementation

  The OAuth implementation currently has zero test coverage, which is a significant security risk.
  Here's the comprehensive test plan:

  1. PKCE Utilities Tests (pkce.test.ts)

  // Test areas:
  - generateCodeVerifier()
    - Generates 43-128 character strings
    - Base64url encoding (no padding, URL-safe)
    - Randomness (different values each time)
  - generateCodeChallenge()
    - SHA256 hashing correctness
    - Base64url encoding of hash
    - Challenge matches expected output for known verifier
  - generateState()
    - 32-character hex string
    - Randomness verification

  2. OAuth Routes Tests (routes.test.ts)

  // Test endpoints:
  - /api/mcp_server/start_oauth
    - Valid provider configuration
    - PKCE parameter generation
    - State storage in pendingOAuthInstalls
    - Cleanup of old pending installs (>10 minutes)
    - Error handling for invalid provider
  - /api/mcp_server/complete_oauth
    - Valid state verification
    - Token exchange with OAuth proxy
    - Token storage in database
    - Environment variable generation
    - Error scenarios (invalid state, expired state, token exchange failure)

  3. Provider Implementation Tests (providers.test.ts)

  // Test provider registry:
  - getOAuthProvider() - returns correct provider
  - hasOAuthProvider() - checks existence
  - Error handling for unknown providers
  // Test each provider:
  - Google: scopes, token patterns, authorization params
  - Slack: token environment variables
  - Slack-browser: browser auth configuration

  4. Browser Authentication Tests (browser-auth-utils.test.ts)

  // Test handleBrowserAuth():
  - Window creation with correct options
  - Navigation rules enforcement
  - Token extraction logic
  - Window cleanup on success/failure
  // Test Slack token extractor:
  - Cookie extraction
  - Local storage extraction
  - Token validation

  5. OAuth Helper Tests (oauth-provider-helper.test.ts)

  // Test functions:
  - validateProvider() - required fields validation
  - getAuthorizationParams() - parameter merging
  - handleProviderTokens() - token mapping and storage

  6. Integration Tests (oauth-integration.test.ts)

  // End-to-end flows:
  - Complete OAuth flow: start → callback → token storage
  - Multiple concurrent OAuth flows
  - Provider-specific flows (Google, Slack)
  // Error recovery:
  - Network failures
  - Invalid tokens
  - Expired sessions

  7. Security Tests (oauth-security.test.ts)

  // Security validations:
  - CSRF protection via state parameter
  - PKCE implementation correctness
  - Token storage security
  - Redirect URI validation
  - Session timeout handling
  - Concurrent request handling

  File Structure

  src/backend/server/plugins/oauth/
  ├── utils/
  │   ├── pkce.test.ts
  │   ├── oauth-provider-helper.test.ts
  │   ├── browser-auth-utils.test.ts
  │   └── slack-token-extractor.test.ts
  ├── providers/
  │   ├── google.test.ts
  │   ├── slack.test.ts
  │   └── slack-browser.test.ts
  ├── routes.test.ts
  ├── providers.test.ts
  └── oauth-integration.test.ts

  Test Implementation Notes

  - Use Vitest (already configured in project)
  - Mock dependencies: OAuth proxy, database, Electron APIs
  - Follow existing test patterns from codebase
  - Include unit and integration tests
  - Cover happy paths and error scenarios
  - Focus on security-critical edge cases

  Priority Order

  1. PKCE utilities - Core security functionality
  2. OAuth routes - Main flow endpoints
  3. Security tests - CSRF, token handling
  4. Provider implementations - Provider-specific logic
  5. Integration tests - End-to-end flows
  6. Browser authentication - Slack-specific functionality

  Running Tests

  pnpm test

  This comprehensive test suite ensures security vulnerabilities are caught early and OAuth flows
  work correctly across all providers before production deployment.
